cloud/auth.go

```diff
@@ -23,23 +23,32 @@ import (
 // Credentials may be either email/password, ssh-based, or a custom token.
 // Upon successful authenticate, the JWT provided by the server is stored in
 // ~/.earthly/auth.jwt, and can be refreshed any time via another call to Authenticate().
-func (c *Client) Authenticate(ctx context.Context) error {
+func (c *Client) Authenticate(ctx context.Context) (AuthMethod, error) {
 	var err error
+	var authMethod AuthMethod
 	switch {
 	case c.email != "" && c.password != "":
+		authMethod = "password"
 		err = c.loginWithPassword(ctx)
 	case c.authCredToken != "":
+		authMethod = "token"
 		err = c.loginWithToken(ctx)
 	default:
+		authMethod = "ssh"
 		err = c.loginWithSSH(ctx)
 	}
```

Comment:
It feels like this is a good opportunity to move this to some form of `authMethod` function/method, e.g.

```go
func (c *Client) authMethod(ctx context.Context) (AuthMethod, error) {
    if c.email != "" && c.password != "" {
        return AuthMethodPassword, c.loginWithPassword(ctx)
    }
    if c.authCredToken != "" {
        return AuthMethodToken, c.loginWithToken(ctx)
    }
    return AuthMethodSSH, c.loginWithSSH(ctx)
}
```

Reply:
good idea!

Reply:
I also totally missed that `authMethod = "password"` wasn't changed to use the consts.